mongo/cursor.go

```diff
@@ -67,6 +68,45 @@ func newEmptyCursor() *Cursor {
 	return &Cursor{bc: driver.NewEmptyBatchCursor()}
 }
 
+// NewMockCursor creates a new Cursor pre-loaded with the provided documents, error and registry. If no registry is provided,
+// bson.DefaultRegistry will be used.
+//
+// The documents parameter must be a slice of documents.
+func NewMockCursor(documents []interface{}, err error, registry *bsoncodec.Registry) (*Cursor, error) {
```

Comment:
Is there a risk that we will need to add options in the future?

If so, we could consider adding a new `CursorOptions` struct to contain the registry:

```
func NewCursor(documents []interface{}, err error, ...options.CursorOptions)
```

That may save another function in case we need to add options in the future.

Reply:
Interesting question. At the moment, there's not much else you can specify on a `Cursor` besides a custom `*session.Client`. I suppose it's possible we'll add fields to `Cursor`, so a new `options.CursorOptions` struct _could_ be helpful in terms of backwards compatibility in the future. I'm not sure it's strictly necessary, though. Any thoughts, @matthewdale ?

[`driver.CursorOptions`](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/batch_cursor.go#L127-L135) already exists and specifies options for a batch cursor. We _could_ allow users to specify custom `driver.CursorOptions` for the underlying batch cursor created with `NewBatchCursorFromDocuments`. I'm not convinced we want to give users the ability to specify things like custom `Crypt` or `ServerAPIOptions` on their `Cursor`s created through `NewCursorFromDocuments`, though.

Reply:
I do not think `driver.CursorOptions` can be exposed directly.

My vote goes for adding a `CursorOptions` struct. I think it is likely cursors will change behavior in the future. Comments in DRIVERS-1857 note the possibility of implementing exhaust cursors in all drivers.

Reply:
I'm hesitant to add an options struct for options that we don't know exist yet. Additionally, making the mock cursor behave exactly the same as a real cursor in all circumstances is explicitly a non-goal because it's impractical (i.e. use a real MongoDB server if you need that).

Additionally, the API change you suggested can be backward compatible if added like:
```
 func NewCursorFromDocuments([]interface{}, error, *bsoncodec.Registry, ...options.CursorOptions) (*Cursor, error)
```
That is a moderately wonky API, but would facilitate the change if necessary.

Reply:
Good points. I relent. No options seems OK given that (1) even if cursors are extended, we do not necessarily need to extend the mock API and (2) we could add a method like ApplyOptions to Cursor if we really needed.

Reply:
Sounds good to me.